plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,388 @@
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) gatherXMS(acc telegraf.Accumulator, url string, wg *sync.WaitGroup,) {
+	defer wg.Done()
+	resp, err := xio.call(url)
+	if err != nil {
+		acc.AddError(err)
+	}
+	tags := map[string]string{
+		"guid":    gjson.Get(resp, "content.guid").Str,
+		"name":    gjson.Get(resp, "content.name").Str,
+		"version": gjson.Get(resp, "content.version").Str,
+		"xms_ip":  gjson.Get(resp, "content.xms-ip").Str,
+	}
+	fields := make(map[string]interface{})
+	fields["xms_write_iops"], err = strconv.Atoi(gjson.Get(resp, "content.wr-iops").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_read_iops"], err = strconv.Atoi(gjson.Get(resp, "content.rd-iops").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_overall_efficiency_ratio"], err = strconv.ParseFloat(gjson.Get(resp, "content.overall-efficiency-ratio").Str, 64)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_ssd_space_in_use"], err = strconv.Atoi(gjson.Get(resp, "content.ssd-space-in-use").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_ram_in_use"], err = strconv.Atoi(gjson.Get(resp, "content.ram-usage").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_ram_total"], err = strconv.Atoi(gjson.Get(resp, "content.ram-total").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_cpu_usage_total"], err = strconv.ParseFloat(gjson.Get(resp, "content.cpu").Raw, 64)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_write_latency"], err = strconv.Atoi(gjson.Get(resp, "content.wr-latency").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_read_latency"], err = strconv.Atoi(gjson.Get(resp, "content.rd-latency").Str)
+	if err != nil {acc.AddError(err)}
+
+	fields["xms_user_accounts_count"], err = strconv.Atoi(gjson.Get(resp, "content.num-of-user-accounts").Raw)
+	if err != nil {acc.AddError(err)}
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) call(endpoint string) (string, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", xio.URL+"/api/json/v3/types/"+endpoint, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	req.AddCookie(xio.Cookie)
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (xio *XtremIO) authenticate() error {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", xio.URL+"/api/json/v3/commands/login?password="+xio.Password+"&user="+xio.Username, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	for _, cookie := range resp.Cookies() {
+		if cookie.Name == "sessid" {
+			xio.Cookie = cookie
+		}
+	}
+	return nil
+}
+
+func init() {
+	inputs.Add("xtremio", func() telegraf.Input {
+		return &XtremIO{Cookie: nil}
```

Comment:
This is the default, skip it
```suggestion
		return &XtremIO{}
```